source/val/validate_memory.cpp

```diff
@@ -389,6 +389,23 @@ spv_result_t ValidateVariable(ValidationState_t& _, const Instruction* inst) {
     }
   }
 
+  // WebGPU: Check that if contains initializer, then storage class is Output,
+  // Private, or Function.
+  if (spvIsWebGPUEnv(_.context()->target_env) && inst->operands().size() > 3) {
```

Comment:
This the same restriction as Vulkan, can you enable (and modify the message) for that environment too please?

Reply:
Done